platform-integration/src/main/java/com/softicar/platform/integration/database/structure/version/DatabaseStructureVersionResource.java

```diff
@@ -0,0 +1,33 @@
+package com.softicar.platform.integration.database.structure.version;
+
+import com.softicar.platform.common.io.resource.container.ResourceSupplierContainer;
+import com.softicar.platform.common.io.resource.supplier.IResourceSupplier;
+import com.softicar.platform.common.io.resource.supplier.IResourceSupplierFactory;
+import com.softicar.platform.common.io.resource.supplier.ResourceSupplierFactory;
+
+/**
+ * Lists JSON-based database structure version definitions, and corresponding
+ * SQL-based migration scripts.
+ * <p>
+ * Any listed file must either be called {@code "vX-structure.json"}, or
+ * {@code "vX-migration.sql"}, with {@code "X"} being a release version number.
+ * The static field names must correspond to the file names (without
+ * extensions).
+ * <p>
+ * By definition, {@code "vX-migration.sql"} contains the necessary statements
+ * to migrate the structure and content of an existing database instance from
+ * version {@code "X-1"} to version {@code "X"}.
+ *
+ * @author Alexander Schmidt
+ */
+@ResourceSupplierContainer
+public interface DatabaseStructureVersionResource {
```

Comment:
why is the name of this class singular? I would expect plural

Reply:
The pluralized name is occupied by the corresponding utility class.
Btw, a search for `ResourceSupplierContainer` references revealed that we currently use singular and plural alike, for these types. So this is not an isolated case.